src/renderer/components/CollabPicker/CollabPicker.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import './CollabPicker.scss';
+import TextField from '../TextField/TextField';
+import Button from '../Button/Button';
+import Select from '@mui/material/Select';
+import { MenuItem } from '@mui/material';
+import { useState } from 'react';
+
+interface Value {
+  username: string;
+  role: number;
+}
+
+interface Options {
+  value: Value;
+  label: string;
+}
+
+interface Props {
+  value?: Value;
+  options?: Options;
+  onChange?: (...args: any[]) => void;
+}
+
+const CollabPicker: React.FC<Props> = ({ options, onChange }) => {
+  [username, onChange] = useState('');
+  const array = [{ role: value?.ro, label: string }];
+  return (
+    <div>
+      <TextField
+        placeholder='Username...'
+        value={name}
+        error={false}
+        onChange={onChange}
+      />
+      <Select label='Role'>
+        {array.map(({ role, label }) => (
+          <MenuItem value={role}>{label}</MenuItem>
+        ))}
+      </Select>
```

Comment:
The `TextField` and `Select` components need specific on change callbacks that modify the state as well as a proper value (based on the local state) to be displayed.

```tsx
// callback that changes state based on the text field input
const handleUsernameChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    setCurrentValue((prev) => {
      const username = event.target.value;
      return { username, role: prev?.role ?? '' };
    });

  // callback that changes state based on selected value
  const handleRoleChange = (event: SelectChangeEvent) =>
    setCurrentValue((prev) => {
      const role = event.target.value;
      return { role, username: prev?.username ?? '' };
    });

  const currentUsername = currentValue?.username ?? '';
  const currentRole = currentValue?.role ?? '';

  return (
    <div>
      <TextField
        placeholder='Username...'
        value={currentUsername}
        onChange={handleUsernameChange}
      />
      <Select label='Role' onChange={handleRoleChange} value={currentRole}>
        {options?.map((option) => (
          <MenuItem key={option.value} value={option.value}>
            {option.label}
          </MenuItem>
        ))}
      </Select>
      <Button>ADD</Button>
    </div>
  );
```